### PR TITLE
gh-554: Handle Draft PRs

### PR DESCRIPTION
**Pull request by itamaro**
1. if opened PR is draft - don't add labels
2. if draft PR is published - add labels as a new non-draft PR
3. if a PR is unpublished - clear labels

### CODE REVIEW

**Comment by Mariatta** [comment]
Thank you!
Can you add test for when a PR went from published to draft?

**Comment by itamaro** [comment]
> Can you add test for when a PR went from published to draft?

Sure!

What's the correct way to mock an existing PR with labels and assert they get deleted?
It looks like the labels are associated with "issue data" based on other tests

**Comment by Mariatta** [comment]
We would test that there is an API call for removing the labels from PR.

**Comment by itamaro** [comment]
updated the PR - is this what you meant? (asserting on `gh.delete_url`)

File: tests/test_stage.py [hunk 1]
```diff
@@ -72,6 +72,74 @@ async def test_stage():
...
+    )
+    await awaiting.router.dispatch(event, gh)
+    assert len(gh.post_) == 0
+    assert (
+        gh.delete_url ==
+        f"https://api.github.com/repos/python/cpython/issues/12345/labels/{encoded_label}"
+    )
+
+
 async def test_opened_pr():
```
Comment (Mariatta) [inline tests/test_stage.py:143]: Can you adjust this test case, the `data` dict should perhaps have `"draft": False` value

Comment (itamaro) [reply tests/test_stage.py:143]: done!

**Review by Mariatta — APPROVED** [review]
Thank you!

**Pull request merged**